### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by dblsaiko**
First commit fixes crash where you remove a part on neighbor update, it crashes (ConcurrentModificationException)  
Second commit fixes you not getting any information except the one stored in the world (meta) when retrieving the IPartSlot in getSlotFromWorld  
Other commits are just some stuff IntelliJ marked as a warning ;)

### CODE REVIEW

File: src/main/java/mcmultipart/api/multipart/MultipartHelper.java [hunk 1]
```diff
@@ -54,7 +54,7 @@ public static boolean addPart(World world, BlockPos pos, IPartSlot slot, IBlockS
     public static Optional<IPartInfo> getInfo(IBlockAccess world, BlockPos pos, IPartSlot slot) {
         return getContainer(world, pos).map(c -> c.get(slot)).orElseGet(() -> {
             if (world instanceof World) {
-                IBlockState state = world.getBlockState(pos);
+                IBlockState state = world.getBlockState(pos).getActualState(world, pos);
```
Comment (asiekierka) [inline src/main/java/mcmultipart/api/multipart/MultipartHelper.java]: getActualState is, as far as I know, primarily used for rendering. Are you sure you're using it correctly?

File: src/main/java/mcmultipart/block/BlockMultipartContainer.java [hunk 2]
```diff
@@ -581,7 +578,7 @@ public void updateTick(World world, BlockPos pos, IBlockState state, Random rand
     }
 
     private void forEach(IBlockAccess world, BlockPos pos, Consumer<PartInfo> consumer) {
-        getTile(world, pos).ifPresent(t -> t.getParts().values().forEach(consumer));
+        getTile(world, pos).ifPresent(t -> new ArrayList<>(t.getParts().values()).forEach(consumer));
```
Comment (asiekierka) [inline src/main/java/mcmultipart/block/BlockMultipartContainer.java]: Wouldn't a nullcheck be more efficient? This *is* a forEach...

**Comment by amadornes** [comment]
I'd suggest PRing things like these separately. It makes no sense to fix a crash in the same commit where you change to some Java 8 specific things. Especially if they're not related in any way shape or form. A PR like this wouldn't be accepted in any kind of serious repo. I'd fix the "big" things in separate PRs and then make another one for minor tweaks in a case like this :P

The reason I'm not pulling this isn't that one, though, because I could just squash everything into a single commit. It's because of the rather hacky things you're doing to fix the CME and fix the actual states.
I know the CME fix works because I've done it before, but it's completely pointless to create a new collection every single time something calls `forEach`. Instead, something like a  `ConcurrentHashMap` should be used, which allows for addition and removal of elements while iterating, and that's what I've ended up implementing.
Now, the `getActualState` commit... I get why you're doing it, but that's not how `World.getBlockState` works, so there is no point in it being like that by default because it would probably break a lot of mods' blocks and nobody would know what exactly was wrong until 5 months later when somebody would run across that bit of code in the API. There's no real need to do that in there and writing a small helper method in your mod should only require a couple of if statements.
I've implemented your J8-related changes (sorting and comparing), though. Not really needed because they don't affect anything other than the aesthetic of the code, but sure, why not :P

TL;DR: I'm not pulling it because of the little hacks. I've fixed the CME with a `ConcurrentHashMap` and I've implemented the minor J8-related tweaks, but not the `getActualState` change because it makes no sense for it to be part of the API.

**Comment by dblsaiko** [comment]
The getActualState was actually the one I needed the most. I need it for determining the part slot in the world. Yes, I'm getting the world and position of the block but these don't mean much since at that position there's a Multipart with potentially multiple of the same parts, so I don't know which exact one is meant (At least I think so, it's been a while since I've implemented this). The slot is determined by a property in the block state, and that property doesn't get saved to the world so I can't determine the part slot.

**Comment by amadornes** [comment]
The good thing about `MultipartHelper` is that it wraps your world even if it's not inside a multipart container, so I think you may be able to get away with just calling `IBlockState#getActualState()` and passing in the world and position if you're certain that it's your block :)

**Comment by dblsaiko** [comment]
Hey, thanks! That worked :P
